spec.md

```diff
@@ -144,18 +144,51 @@ circuit MyTop...
 
 ## Circuits
 
-All FIRRTL circuits consist of a list of modules, each representing a hardware
-block that can be instantiated. The circuit must specify the name of the
-top-level module.
+A FIRRTL circuit is a named collection of FIRRTL modules.  Each module is a
+hardware "unit" that has ports, registers, wires, and may instantiate other
+modules (see: [@sec:modules]).  (This is the same concept as a Verilog
+`module`{.verilog}.)  A public module is usable outside of the current circuit.
```

Comment:
what does it mean to be "usable". Should we say that a public module follows the defined verilog ABI? And that it blocks constant propagation across its boundaries? I'm still unclear if that is what "public" means.

Reply:
i guess you clarify this below

Reply:
Talking about what it means to be "public" is difficult here as we haven't even introduced module, yet. This is more of rewriting the spec to put things in a more logical order. The requirements of `public` are specified in the "Public Modules" subsection of the "Module" section which follows the "Circuit" section. @mmaloney-sf has brought up the fact that this whole thing is hard to read and it may make sense to reverse all these sections. I.e., introduce `module` before `circuit`. This type of thing is largely out of scope of this PR. I agree we should do some big rewrites, though.